### PR TITLE
Upgrade to EventStoreDB 21.2 and use gRPC client. Fixes #36

### LINKED ISSUE #36: Upgrade domain-lib to use EventStoreDb 21.2 and use the gRPC client

### PR DESCRIPTION
**Pull request by jonclare**

### CODE REVIEW

File: src/DomainLib.EventStore.AspNet/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -15,17 +14,17 @@ public static class ServiceCollectionExtensions
...
         {
-            AddEventStoreAggregateRepository<object, object>(services, configuration, aggregateRegistry);
+            return AddEventStoreAggregateRepository<object, object>(services, configuration, aggregateRegistry);
         }
 
         /// <summary>
         /// Adds an aggregate repository that uses EventStoreDb to persist events 
         /// </summary>
-        public static void AddEventStoreAggregateRepository<TCommandBase, TEventBase>(this IServiceCollection services,
+        public static IServiceCollection AddEventStoreAggregateRepository<TCommandBase, TEventBase>(this IServiceCollection services,
```
Comment (carlosrfernandez) [inline src/DomainLib.EventStore.AspNet/ServiceCollectionExtensions.cs:27]: Why the change of signature?

Most service collection extension methods do it like how you had it before 😛

Comment (jonclare) [reply src/DomainLib.EventStore.AspNet/ServiceCollectionExtensions.cs:27]: I've seen quite a few that return the service collection and allow you to chain them.
Figured I'd go for that pattern

File: src/DomainLib.Persistence.EventStore/EventStoreEventsRepository.cs [hunk 2]
```diff
@@ -57,107 +56,76 @@ public async Task<long> SaveEventsAsync<TEvent>(string streamName, long expected
...
-                        if (sliceStartPosition >= eventDatas.Length)
-                        {
-                            eventsToSave = false;
-                        }
-                    }
-
-                    writeResult = await transaction.CommitAsync();
-                    Log.LogDebug("Transaction {TransactionId} committed. Write Id {WriteId}", transaction.TransactionId, writeId);
-                }
+                writeResult = await _client.AppendToStreamAsync(streamName, streamVersion, eventDatas);
```
Comment (carlosrfernandez) [inline src/DomainLib.Persistence.EventStore/EventStoreEventsRepository.cs:67]: This seems to have been extremely reduced and a lot more simple now?

I'm assuming all the batching/transactional stuff is either not needed or done internally by the _client?

**Review by carlosrfernandez — APPROVED** [review]
LGTM. I still want to get a bit more context overall on the lib, but this looks good.

File: src/DomainLib.Persistence.EventStore/EventStoreEventsRepository.cs (hunk 2 above)
Comment (jonclare) [reply src/DomainLib.Persistence.EventStore/EventStoreEventsRepository.cs:67]: Yep, it exposes an IAsyncEnumerable now and does everything through that. No batching on our side.
One upshot of that is that loading lots of events is significantly slower. I couldn't find a better way to do that.
Might raise an issue with EventStore?
